lua/spec/neotest-rspec/init_spec.lua

```diff
@@ -0,0 +1,17 @@
+local plugin = require('neotest-rspec')
+
+describe('is_test_file', function()
+  it('matches rspec files', function()
+    assert.equals(true, plugin.is_test_file('./spec/foo_spec.rb'))
+  end)
+
+  it('does not match plain ruby files', function()
+    assert.equals(false, plugin.is_test_file('./lib/foo.rb'))
+  end)
+end)
+
+describe('discover_positions', function()
+  it('matches rspec files', function()
+    assert.equals('something', plugin.discover_positions('./spec/nested_tests/nested_spec.rb'))
```

Comment:
Currently fails with the following error:

```
neotest/lua/neotest/lib/treesitter/init.lua:138: attempt to yield across C-call boundary
```

Seems to be caused when neotest attempts to sleep:

https://github.com/nvim-neotest/neotest/blob/f26979786377704ce85fe43d453418249b1fe7e4/lua/neotest/lib/treesitter/init.lua#L137-L141

Maybe this could be mocked out.

Reply:
This was the biggie I hit...I wonder if we need to utilise `plenary.async.tests`. I did it recently for my [session plugin](https://github.com/olimorris/persisted.nvim/blob/main/tests/autoload/autoload_allowed_dir_spec.lua). So you call `async.describe` instead. Let me know if that does anything.

Reply:
The async helpers works and I was able to get access to the neotest `Tree` table.